### ISSUE TITLE
Exporter name registry

### ISSUE THREAD

**Issue by alexweggs** [FriendsOfSylius/SyliusImportExportPlugin#220]
Hi,
I've got an issue with custom entity.
I try to configure a basic exporter for my entity "seo_url".

I configured the services as in the documentation but when i launched the export command i've got this error :
 
**There is no 'sylius.seo_url.csv' exporter.**  

In the displayed list i've got '**seo_url**' as an available exporter.

When i dump the exportRegisrty i see that the expected name is 'app.seo_url.csv' and not 'sylius.seo_url.csv' 

Here is the cmd line : 

bin/console sylius:export seo_url seo_urls.csv --format=csv

And the configuration : 

    # seo_url CSV exporter
    sylius.exporter.seo_urls.csv:
        class: FriendsOfSylius\SyliusImportExportPlugin\Exporter\ResourceExporter
        arguments:
            - "@sylius.exporter.csv_writer"
            - "@sylius.exporter.pluginpool.seo_urls"
            - ["Id", "Url" ,"Title"]
            - "@sylius.exporters_transformer_pool" # Optional
        tags:
            - { name: sylius.exporter, type: app.seo_url, format: csv }

    # PluginPools for seo exporter
    sylius.exporter.pluginpool.seo_urls:
        class: FriendsOfSylius\SyliusImportExportPlugin\Exporter\Plugin\PluginPool
        arguments:
            - ["@sylius.exporter.plugin.resource.seo_url"]
            - ["Id", "Url" ,"Title"]

        # Plugins for seo_url exporter
    sylius.exporter.plugin.resource.seo_url:
        class: FriendsOfSylius\SyliusImportExportPlugin\Exporter\Plugin\ResourcePlugin
        arguments:
            - "@app.repository.seo_url"
            - "@property_accessor"
            - "@doctrine.orm.entity_manager"

Maybe i miss something ;)

Thanks a lot !

**Comment by oallain** [comment]
Hi @alexweggs 

You are currently required to use a resource type with a **sylius** prefix.

```
sylius.exporter.seo_urls.csv:
    class: FriendsOfSylius\SyliusImportExportPlugin\Exporter\ResourceExporter
    arguments:
        - "@sylius.exporter.csv_writer"
        - "@sylius.exporter.pluginpool.seo_urls"
        - ["Id", "Url" ,"Title"]
        - "@sylius.exporters_transformer_pool" # Optional
    tags:
-        - { name: sylius.exporter, type: app.seo_url, format: csv }
+        - { name: sylius.exporter, type: sylius.seo_url, format: csv }
```

I hope this will help you.

**Comment by alexweggs** [comment]
@oallain: thnaks :) but this is eaxcly the problem, so you mean yan can't use it with custom resources defined in sylius_resources.yaml ?

**Comment by oallain** [comment]
Hi, @alexweggs 

I made a PR #221  that works well for me, hoping that it meets your expectations.

**Comment by alexweggs** [comment]
Hi @oallain 
I wiil try it as soon as i can, thanks a lot for that !

**Comment by alexweggs** [comment]
@oallain , seeme to work very well :)

Any tips to add the import/export action on the custom entoty admin page list ?

**Comment by oallain** [comment]
Hi @alexweggs ,

I try it with #221 and i got export buton and import form in my custom ressource list.
Not you ?

**Comment by alexweggs** [comment]
Hi @oallain ,

Nope, i can export via command line but no button on my custom ressource list !

**Comment by oallain** [comment]
Hi @alexweggs 

Ho wan i reproduce it ?
You try tu export data from https://github.com/stefandoorn/SyliusSeoUrlPlugin ?
Can you copy/paste your configuration of exporter ?

**Comment by alexweggs** [comment]
Hi @oallain ,

I use custom seo entity, with custom datas so i need to do my own exporter.
You have the conf on first post ;)

**Comment by oallain** [comment]
Hi @alexweggs 

Sorry for long time response.
Have you already find a fix ?
If not, you can find how link is buid in :https://github.com/FriendsOfSylius/SyliusImportExportPlugin/blob/75e7ad39be62cb4e7320a5a8c2191a686145d47f/src/Listener/ExportButtonGridListener.php#L79

**Comment by krystal25** [comment]
Hi @oallain,

I am facing the same problem with my custom entity - no export button in the admin. I added a different button by defining a bulk action of the grid, and the export works fine, except for filtered data. I noticed that it should go in the `onSyliusGridAdmin()`, but it doesn't, so I assume there should be an event implemented? This `if ($container->has($eventHookName))` from `registerSingleEventListenerForExportButton` returns false, `$eventHookName` being `app.grid_event_listener.admin.crud_app.custom_csv_export`. How should I inject this event in my container?
